xbmc/video/ContextMenus.cpp

```diff
@@ -58,6 +58,8 @@ bool CMarkWatched::IsVisible(const CFileItem& item) const
   {
     if (item.HasVideoInfoTag())
       return item.IsVideoDb();
+    else if (item.GetProperty("IsVideoFolder").asBoolean())
```

Comment:
I thought that this context class is only involved if the item is holds a video item, because it's `video/ContextMenus.cpp`. If that is not the case ignore my comment.